WPMedia/fusion-news-theme-blocks
# Tbrands 27 algolia assortment content source

## Description

Add algolia content source

## Jira Ticket

- [TMEDIA-27](https://arcpublishing.atlassian.net/browse/TMEDIA-27)

## Acceptance Criteria

A new block in the arc-themes-blocks repo - algolia-assortment-content-source

A Fusion Content Source that can query the Algolia Search API with the following parameters

Index

Context - Algolia Param - ruleContexts

Filters - Algolia Param - filters

Query - Algolia Param - query

Size - Algolia Param - hitsPerPage - default = 20

Page - Algolia Param - page - default = 0

The content source relies on ApplicationId and APIKey to set up the Algolia search methods

Content Source has TTL set to the Fusion minimum of 2 minutes

## Test Steps

1. Go to feature pack repo 
2. Checkout this branch `git checkout TBRANDS-27-algolia-assortment-content-source}`
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/algolia-assortment-content-source-block`
4. Go to blocks repo
5. Checkout this branch `git checkout TBRANDS-27-algolia-assortment-content-source}`
6. Go to feature pack repo.  https://github.com/WPMedia/arc-themes-feature-pack/pull/218
7. Ensure your .env has plaintext values for `algoliaSearchKey` and `algoliaAppId`. See https://github.com/WPMedia/arc-themes-feature-pack/pull/209#issue-1154629794 for getting these values if you don't have them already
8. Make sure you've added `@local` to the block name in the blocks.json
9. Run `npx fusion start -f -l @wpmedia/algolia-assortment-content-source-block`
10. Go to content debugger on local for algolia assortment http://localhost/pagebuilder/tools/debugger?_website=arc-commerce&type=content-debugger&source=algolia-assortment
11. Run a query like and notice the first image url. Notice there are 20 items returned. This means there are 20 items by default and it starts on page 0. `searchdemoproduct_commercedev_` for index, `shoes` for query, `featured_shoes` for rule contexts
![Screen Shot 2022-04-12 at 10 57 59](https://user-images.githubusercontent.com/5950956/163004309-79c4ae73-119b-4491-8c28-34a4d736520a.png)
12. Now set the same query to 21 count. You'll see 21 items
13. Now set the page to 1. You will see a different image url than by default before
![Screen Shot 2022-04-12 at 10 59 51](https://user-images.githubusercontent.com/5950956/163004842-d8dafcb6-bf8d-4327-8ebf-bd4ab70de549.png)
14. Now add a filter to notice the effect of adding a specific sku. (In our example, there's 3 items with the same sku.) The filter text should be: `("sku": "SHOE34YXGQYWM45Y")`
![Screen Shot 2022-04-12 at 11 01 49](https://user-images.githubusercontent.com/5950956/163005190-87a2fc63-fb36-4ead-9486-6458e7d9e382.png)

## Effect Of Changes

### Before

- No algolia assortment content source

### After

- have an algolia assortment content source! 

## Dependencies or Side Effects

- Need the environment files. I've added them in the feature pack pr for the dev commerce environment with the key encrypted
- Merge and deploy https://github.com/WPMedia/arc-themes-feature-pack/pull/218

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



### Notes 
- Go to https://github.com/WPMedia/arc-themes-feature-pack/pull/209#issue-1154629794 for getting the algoliaSearchKey set in `.env`. The `algoliaAppId` should also be set in the `.env` 
- Add     "@wpmedia/algolia-assortment-content-source-block@local" to blocks.json. run `npx fusion start -f -l  @wpmedia/algolia-assortment-content-source-block` to get this working locally
- I've added additional comments after feedback from @marissa-halpert 

Working example with options set from ticket 

index
searchdemoproduct_commercedev_

<img width="648" alt="Screen Shot 2022-04-11 at 14 05 05" src="https://user-images.githubusercontent.com/5950956/162813152-bda4940f-07b4-4ffe-8da9-95af01bdd372.png">



### Questions
- I don't seem to be able to use filters. I've tried various configurations https://www.algolia.com/doc/api-reference/api-parameters/filters/ , including is_fk_advantage_product:false and  "is_fk_advantage_product":"false". Is there something that I'm missing with the format or the data? Looking for help on this end as I did not see it in Nick's pr 
- For the algolia app id, I set this in the environment file. I'm assuming this is ok based on the ticket ac https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TBRANDS-27 Is this ok? 


### Todos

- [x] Update readme with basic info and params
- [x] Create feature pack pr with the environment values, including encrypted search key and plaintext app id
- [x] Add tests with mocking algolia api
